core/src/main/java/org/elasticsearch/transport/netty/MessageChannelHandler.java

```diff
@@ -285,6 +261,11 @@ protected String handleRequest(Channel channel, StreamInput buffer, long request
         return action;
     }
 
+    protected void validateRequest(Marker marker, StreamInput buffer, long requestId, String action) throws IOException {
+        marker.validateRequest(buffer, requestId, action);
```

Comment:
By itself, the method is useless, right. But previously `NettyTransportIT`had a custom `MessageChannelHandler` implementation which had to reimplement `#handleRequest()` almost completely just to inject custom errors in the test (and obviously the implementation already deviated from production code). By introducting this method, it is possible to override request validation much more fine-grained and still inject custom error detection in test code at the right place.